harness/determined/common/storage/base.py

```diff
@@ -75,31 +78,20 @@ def restore_path(self, storage_id: str) -> Iterator[str]:
         """
         pass
 
-    def delete(self, storage_id: str) -> None:
-        """
-        Delete the stored data from persistent storage.
-        """
-        storage_dir = os.path.join(self._base_path, storage_id)
-
-        check_true(
-            os.path.exists(storage_dir), "Storage directory does not exist: {}".format(storage_dir)
-        )
-        check_true(
-            os.path.isdir(storage_dir), "Storage path is not a directory: {}".format(storage_dir)
-        )
+    @abc.abstractmethod
+    def upload(self, src: str, dst: str) -> None:
```

Comment:
I feel like this API makes it hard to distinguish which parameters are concrete local filesystem paths, and which are "remote" destinations which sometimes may be paths and sometimes may look like paths but really aren't, such as s3 key names or prefixes.

I understand this may be a bit of work but we could solve it through typing, in particular have `Union[str, Path]` for concrete local fs paths.

so upload will become
```suggestion
    def upload(self, src: Union[str, Path], dst: str) -> None:
```

For return values, returning `pathlib.Path` for things that are paths also may be helpful in a similar vein. This applies for `restore_path` which may return `-> Iterator[Path]`.

Reply:
done, but I used `Union[str, os.PathLike]` instead of `Union[str, pathlib.Path]`.